### DISCUSSION TITLE
go back in time

### DISCUSSION THREAD

**Discussion by JoostvanBruggen** [Mattk70/Chirpity-Electron#171] [Q&A]
Many times when I quickly scroll through the recorded night counting the Redwings I see something in the spectrogram wich I find interesting. But often I already clicked 2-3 time and it is out of view. I can go back because there is no option to scroll back quickly, only very slowly with the back-arrow button ( I think).

I would be nice to go back a lot faster to recapture a spectrogram and listen to it.

Joost van Bruggen, Holland

**Comment by Mattk70** [comment]
Hi Joost,

You can scroll back as fast as forward, The various ways to navigate the timeline are described in the keyboard help. You don't say *what* you clicked to move forward, but you can use PageUp as opposed to PageDown to go back a screen at a time instead of forward, Shift-Tab instead of Tab to move back though previous results, and with the latest version you can use the mouse wheel to go forward and back as well (while the mouse is over the spectrogram)

**Comment by ltgustavsen** [comment]
I just wanted to mention that I miss the ability to scroll back to the previous file as well. I usually log 10-minute files, open the folder, and when I scroll and reach the end of a file—which happens quite quickly—I can't continue scrolling to the next file.

**Reply by Mattk70** [reply]
@ltgustavsen - 

I see a lot of people that use Chirpity like it was Audacity or something like it. They open a file and scroll through the audio, and when they get to something interesting, they'll highlight and analyse that selection of audio. I've always thought this was because they'd carried over their workflow patterns from the "Audacity approach".

The idea with Chirpity is that you analyse a complete file or collection of files, and then review the detections. The scrolling feature is really designed as a way to check the context of a detection. That is, you scroll around a detection to see if there was a similar call up to a minute before or after.

I also wonder why, if you are recording over long periods of time, you choose to split the audio files every ten minutes?

Am I missing something here, can you help me understand the reason for your workflow?

**Reply by ltgustavsen** [reply]
I can't remember why I set my audio recorder to log in 10-minute intervals. I think it was to avoid losing an entire hour of recording if the battery died or something similar. Anyway, I plan to extend the logging time in the future.

My purpose for logging is mostly out of curiosity and to locate birds I can photograph. My workflow is probably similar to what you described. I log both day and night. I open the entire catalog and look through the hits for interesting birds. Sometimes, if there's something more remarkable, I browse around to get the context as you described. I often mark areas to run analysis to see what the nearby sounds are because they aren't tagged. If I'm in timeline mode and haven't selected a bird species, it's a bit easier since I can see the hit in the hit list. But typically, I select a bird species, and then information about nearby hits of other species disappears. So, I move around as you described, but I've had too short files and quickly hit the limitation that the next file isn't loaded. As mentioned, I will increase the logging time.

Further in my workflow, I play around with the sound optimization settings to play the sounds as best as possible. When I do this, it's a bit annoying that the settings window closes every time I play a sound.

I've used Audacity a lot for noise removal and such, but I'm using it less and less because the result often sounds good, but when I run the exported file back in Chirpity, I see that I've lost a lot of information. I also think what you've achieved with audio optimization via ffmpeg is outstanding work. On the other hand, the exported Audacity labels are handy for seeing birds in context.

**Reply by Mattk70** [reply]
@ltgustavsen - thanks for the explanation. It would be a very big rewrite for me to display multiple detections on the spectrogram. It would be easier to remember the detection you'd selected and keep that selection when you remove the species filter. Then you would have the previous and next detection regardless of the species next to the one you had selected. As for smoothly scrolling to the next file, I haven't looked into it.

> Further in my workflow, I play around with the sound optimization settings to play the sounds as best as possible. When I do this, it's a bit annoying that the settings window closes every time I play a sound.

It's *much* quicker to use the spacebar to play/pause. If you have a region highlighted, you can press 'p' to play and replay the region. With either keyboard approach, the panel won't close.

> when I run the exported file back in Chirpity, I see that I've lost a lot of information

This could be due to your Audacity audio export settings, too much compression?

**Reply by ltgustavsen** [reply]
> > Further in my workflow, I play around with the sound optimization settings to play the sounds as best as possible. When I do this, it's a bit annoying that the settings window closes every time I play a sound.
> 
> It's _much_ quicker to use the spacebar to play/pause. If you have a region highlighted, you can press 'p' to play and replay the region. With either keyboard approach, the panel won't close.

Thank you for this. I wasn’t able to figure this out before. The reason was that I didn’t understand that the focus needed to be moved out of the controller with the mouse. Now that I understand it, it works well.
I’m trying to upload a screen recording  in case others haven’t understood this either. 

https://github.com/user-attachments/assets/763312f6-afa5-43cc-9cad-61ba51248ec4

**Reply by Mattk70** [reply]
I think I will make it so the form elements for audio preferences don't capture focus, that would be better.
